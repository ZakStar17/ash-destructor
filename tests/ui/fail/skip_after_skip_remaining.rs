use ash_destructor::DeviceDestroyable;

#[path = "../../utils/mod.rs"]
mod utils;

use utils::ImplDeviceDestroyable;

#[derive(DeviceDestroyable)]
struct Named {
    a: ImplDeviceDestroyable,
    #[skip]
    #[skip_remaining]
    b: String,
    c: u32,
    #[skip]
    d: bool,
}

#[derive(DeviceDestroyable)]
struct Unnamed(#[skip_remaining] ImplDeviceDestroyable, #[skip] String);

fn main() {}
