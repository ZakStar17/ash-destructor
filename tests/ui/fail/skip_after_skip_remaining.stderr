error: Attribute #[skip] is not allowed after a #[skip_remaining] attribute declaration
  --> tests/ui/fail/skip_after_skip_remaining.rs:11:5
   |
11 |     #[skip]
   |     ^^^^^^^

error: Attribute #[skip] is not allowed after a #[skip_remaining] attribute declaration
  --> tests/ui/fail/skip_after_skip_remaining.rs:15:5
   |
15 |     #[skip]
   |     ^^^^^^^

error: Attribute #[skip] is not allowed after a #[skip_remaining] attribute declaration
  --> tests/ui/fail/skip_after_skip_remaining.rs:20:57
   |
20 | struct Unnamed(#[skip_remaining] ImplDeviceDestroyable, #[skip] String);
   |                                                         ^^^^^^^
